native/src/widget/pick_list.rs

```diff
@@ -168,50 +184,63 @@ where
             .pad(self.padding);
 
         let text_size = self.text_size.unwrap_or(renderer.default_size());
+        let menu_text_size = self.menu_text_size.unwrap_or(text_size);
         let font = self.font;
+        let menu_font = self.menu_font.unwrap_or(font);
+
+        let measure = |label: &str, text_size, font| -> u32 {
+            let (width, _) = renderer.measure(
+                label,
+                text_size,
+                font,
+                Size::new(f32::INFINITY, f32::INFINITY),
+            );
 
-        let max_width = match self.width {
+            width.round() as u32
+        };
+        let placeholder_width = self
+            .placeholder
+            .as_ref()
+            .map(String::as_str)
+            .map(|s| measure(s, text_size, font))
+            .unwrap_or(100);
+        let selected_width = match self.width {
             Length::Shrink => {
-                let measure = |label: &str| -> u32 {
-                    let (width, _) = renderer.measure(
-                        label,
-                        text_size,
-                        font,
-                        Size::new(f32::INFINITY, f32::INFINITY),
-                    );
-
-                    width.round() as u32
-                };
-
-                let labels = self.options.iter().map(ToString::to_string);
-
-                let labels_width =
-                    labels.map(|label| measure(&label)).max().unwrap_or(100);
-
-                let placeholder_width = self
-                    .placeholder
-                    .as_ref()
-                    .map(String::as_str)
-                    .map(measure)
+                let selected = self.selected.as_ref().map(ToString::to_string);
+                let selected_width = selected
+                    .map(|label| measure(&label, text_size, font))
                     .unwrap_or(100);
 
-                labels_width.max(placeholder_width)
+                selected_width.max(placeholder_width)
             }
             _ => 0,
         };
+        let max_width = {
+            let labels = self.options.iter().map(ToString::to_string);
+
+            let labels_width = labels
+                .map(|label| measure(&label, menu_text_size, menu_font))
+                .max()
+                .unwrap_or(100);
+
+            labels_width.max(placeholder_width)
+        };
```

Comment:
This change will make `layout` hit the worst case and measure all the labels independently of the `PickList` configuration. Thus, a `PickList` with thousands of labels would cause a considerable slow down.

Instead, we should probably measure the labels when opening the `Menu` while also making sure the bounds don't overflow the screen edges.

Also, maybe we should keep the current behavior as the default one and make the new one opt-in with an additional configuration enum?

Reply:
Yeah, this should probably be opt-in behaviour. As for measuring the labels when opening the `Menu`, that should be done in `Menu` layout itself then I guess?

Reply:
Yes, but only when strictly necessary (i.e. when the `PickList` does not provide a predefined `width` for the `Menu`).